pkg/deploy/controller/deployment/factory.go

```diff
@@ -110,6 +111,10 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 					return false
 				}
 				if retries.Count > 1 {
+					deployment, ok := obj.(*kapi.ReplicationController)
+					if ok {
+						eventRecorder.Eventf(deployment, kapi.EventTypeWarning, "FailedRetry", "About to stop retrying %s: %v", deployment.Name, err)
```

Comment:
May want to include in the message what the controller is doing - "Unable to update deployment N for dc_name, will retry later: %v" or similar to give users a hint